app/AvailablePlugin/EmailAddressWidget/View/CoEmailAddressWidgets/display.ctp

```diff
@@ -88,7 +94,21 @@
         let xhr = callRegistryAPI(url, 'GET', 'json', this.setEmailAddresses, '', this.generalXhrFailCallback);
       },
       setEmailAddresses: function(xhr){
-        this.emailAddresses = xhr.responseJSON.EmailAddresses;
+        let allEmailAddresses = xhr.responseJSON.EmailAddresses;
+        // limit the email addresses to the emailType:
+        let emailAddressesOfType = [];
+        allEmailAddresses.forEach((address) => {
+          if(address.Type == this.core.emailType) {
+            emailAddressesOfType.push(address);
+          }
+        });
+        // Test to see if we've reached the max count of email addresses for this widget
```

Comment:
this could be simplified to this:
```
 this.core.emailLimitReached = Number(this.core.emailLimit) > 0 && emailAddressesOfType.length >= Number(this.core.emailLimit);
```

Reply:
Done :)